src/libAtomVM/ets_hashtable.c

```diff
@@ -130,7 +131,12 @@ EtsHashtableErrorCode ets_hashtable_insert(struct EtsHashTable *hash_table, stru
     struct HNode *node = hash_table->buckets[index];
     struct HNode *last_node = NULL;
     while (node) {
-        if (term_compare(key, node->key, TermCompareExact, global) == TermEquals) {
+        TermCompareResult cmp = term_compare(key, node->key, TermCompareExact, global);
+        if (UNLIKELY(cmp == TermCompareMemoryAllocFail)) {
+            return EtsHashtableError;
```

Comment:
It is ok to raise a specific error about out of memory.
On microcontrollers it is not an unlikely event, so it is better to keep it clear when it happens.

Reply:
Renamed to `EtsHashtableOutOfMemory`. We cannot raise here.